index.js

```diff
@@ -55,47 +42,67 @@ const loadConfiguration = async file => {
 
         return config;
     } catch (err) {
-        console.error(err.message)
-        exit(1);
+        onError(err.message)
     }
 }
 
-const generate = async ({ output, config }) => {
-    config = await loadConfiguration(config);
+const generate = async ({output, config, language}) => {
+    await loadConfiguration(config);
+
+    const clazz = languageClasses[language];
+    if (!clazz) {
+        onError(`Language must be one of [${Object.keys(languageClasses)}]`);
+    }
+    const lang = new clazz();
 
     /** @type {Array<object>} collections*/
     const collections = (await database.listCollections()).collections;
-    const types = [];
+    const types = buildCollectionTypes(collections)
+
+    if (!existsSync(output)) {
+        await mkdir(output, {recursive: true});
+    }
+
+    for (const type of types) {
+        const content = lang.generate(type)
+
+        try {
+            const destination = path.join(output, `./${type.name}${lang.getFileExtension()}`);
+            await writeFile(destination, content);
+            console.log(`Generated ${destination}`)
+        } catch (err) {
+            console.error(err.message)
+        }
+    }
+}
+
+const buildCollectionTypes = (collections) => {
+    let types = [];
     collections.forEach(collection => {
         types.push({
-            name: toPascal(collection.name),
+            name: collection.name,
             attributes: collection.rules.map(rule => {
                 return {
                     name: rule.key,
-                    type: toType(rule.type),
+                    type: rule.type,
                     array: rule.array,
                     required: rule.required
                 }
+            }).sort((a, b) => {
+                return (a.required === b.required)
+                    ? 0
+                    : a.required
+                        ? -1
+                        : 1;
```

Comment:
Do we wanna interfere with the natural order from Appwrite for beauty reason? 🤔 

I like it 😄